### ISSUE TITLE
Replace jdepend-maven-plugin with code-assert

### ISSUE THREAD

**Issue by dbwiddis** [oshi/oshi#1026]
The [jdepend-maven-plugin](https://www.mojohaus.org/jdepend-maven-plugin/) was last published in 2015 and does not support JDK8.
> [INFO] Generating "JDepend" report       --- jdepend-maven-plugin:2.0:generate-no-fork
>    Unknown constant: 18
>   (multiple repeats)

It appears the [project was forked](https://github.com/nidi3/jdepend) and maintained up to 2.9.1 but that fork is now deprecated.  The new suggested replacement is [code-assert](https://github.com/nidi3/code-assert).

We should remove jdepend and consider whether we want to add code-assert.

**Comment by hazendaz** [comment]
Thanks.  I'll take a look as soon as I can.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by dbwiddis** [comment]
It looks useful... many "style" errors (such as javadoc `@param` mismatches or broken `@link`s) I don't catch unless I closely watch maven build logs.  Appveyor tends to catch those but it's post-commit and requires digging through the log reports as well.  Running tests from my IDE is a much better place to find/fix those issues.

Don't need the PMD support though.  :)

**Comment by dbwiddis** [comment]
Well, I removed jdepend.  Code-assert is easy enough to add, but there isn't much in there that I want to test yet (as in, all the tests failed for many reasons that I don't want to change).   I think I'm just going to leave it as is... I know I have the ability to test things later if I want to add it.

**Comment by hazendaz** [comment]
jdepend and code-assert don't match really.  While they note basing there code on it, it sounds like duplication and looks like you have to physically code out in unit tests which is a bit much.  jdepend is to show how things (objects) come together.  It doesn't show issues.  No issue on removing it as value is little (at least for me, I cannot ever make heads or tails out of what it's telling me anyways).  Code-assert though is an aggregate of many plugins which violates maven standards (one plugin doing one thing well).  Also, its tied to obsolete stuff like findbugs which I'm the maintainer of for maven under it's new branding as spotbugs for many years now.  All the example usage is code driven from what I can tell which to me doesn't make much sense when all those items are maven plugins in their own right that just are configured.  I suspect all are already in the poms here.  I don't recall off hand but likely I put them in at some point but don't cause failures if things are not exactly correct.  I'm not sure I would go as far as writing tests to assert those conditions though.  However, when you get around to doing some of that, please call me out as I would like to see what it looks like and if my mind is changed :)  I'm always open for new stuff, just sometimes need that extra push to see the value :)

**Comment by hazendaz** [comment]
btw- if I did add all those other plugins before along the way, just run 'mvn site' and look at the resulting site.  It would have many issues listed :)  

Also if you want something more in line to prevent bad usage, error prone from google is great but it lacks support properly for jdk 12 and non existent still on jdk 13+

**Comment by hazendaz** [comment]
https://github.com/google/error-prone

And setup you can see in my super pom -> https://github.com/hazendaz/base-parent/blob/master/pom.xml

**Comment by hazendaz** [comment]
Back on the original issue of jdepend support. I did send maven a PR to get to 2.9.5 of jdepend (ie from this guy that marked it obsolete). [here](https://github.com/mojohaus/jdepend-maven-plugin/pull/5).  Not sure if that will get merged or not as the plugin itself might also be dead at this point.

**Comment by dbwiddis** [comment]
Yep, I'm familiar with the "site" and the report jdepend used to produce (and still does in the 3.X branch with Java 7).  As you say it's not that useful.  I also did pretty much go through every conceivable iteration of code-assert's options and most of it (and a lot of our "site" frankly) duplicates Sonarqube reports.  I did think that I might want to include some of the javadoc tests some unknown day in the future, but for now, it's not a priority.
